src/odb/include/odb/db.h

```diff
@@ -5330,6 +5330,27 @@ class dbSite : public dbObject
   ///
   bool getSymmetryR90();
 
+  ///
+  /// set the row pattern of this site
+  ///
+  void setRowPattern(
+      std::vector<std::pair<std::string, std::string>>& row_pattern);
```

Comment:
This should mirror getSitePattern and take a "const std::vector<std::pair<dbSite*, dbOrientType>>&" argument.  The conversion from strings should happen in lefin.